quark-test.c

```diff
@@ -436,6 +436,103 @@ t_fork_exec_exit(const struct test *t, struct quark_queue_attr *qa)
 	return (0);
 }
 
+static int
+t_process_iterator(const struct test *t, struct quark_queue_attr *qa)
+{
+	struct quark_queue		 qq;
+	struct quark_process_iter	 qi;
+	const struct quark_process	*qp;
+	int				 n_children = 3;
+	pid_t				 children[n_children];
+	struct args			*args;
+	size_t				 expected_len;
+	int				 i;
+	int				 j;
+	char				 cwd[PATH_MAX];
+
+	if (quark_queue_open(&qq, qa) != 0)
+		err(1, "quark_queue_open");
+
+	quark_process_iter_init(&qi, &qq);
+
```

Comment:
Here you initialize the iterator before drain_for_pid, so the iterator can be invalidated, just move this to before the  main loop. I realize this is not ideal, maybe I should change iter_init to point to NULL, instead of pointing to RB_ROOT.